### PR TITLE
Move downloader to `internal` package

### PR DESCRIPTION
**Pull request by peterjan**
This PR moves the `downloader` to our `internal` package, no changes to the code itself. This way it's consistent with the uploader and we're setting ourselves up to move migrations into the `bus` 's autopilot.

Related https://github.com/SiaFoundation/renterd/issues/1646

### CODE REVIEW

**Comment by peterjan** [comment]
Failing on an NDF but since it's moving files around I'm pushing it into review.

File: internal/download/downloader/downloader_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	t.Run("stop after enqueue", func(t *testing.T) {
+		hm := mocks.NewHostManager()
+		dl := New(context.Background(), hm.Downloader(api.HostInfo{}))
+		req := SectorDownloadReq{
+			Ctx:   context.Background(),
+			Resps: NewSectorResponses(),
+		}
+
+		dl.Enqueue(&req)
+		dl.Stop(errors.New(t.Name()))
```
Comment (ChrisSchinnerl) [inline internal/download/downloader/downloader_test.go:49]: I'm surprised this has never caused NDFs considering the download might be done before `Stop` is called.

**Review by ChrisSchinnerl — COMMENTED** [review]
This PR seems to be pretty independent from the RHP4 integration one right? If so I'd merge it first and rebase the other PR

File: internal/download/downloader/downloader_test.go (hunk 1 above)
Comment (peterjan) [reply internal/download/downloader/downloader_test.go]: The downloader is not processing its queue.

**Comment by peterjan** [comment]
> This PR seems to be pretty independent from the RHP4 integration one right? If so I'd merge it first and rebase the other PR

I rebased it. It was independent but it did overlap and it will cause conflicts on `pj/rhp4-downloads` after being merged. No big deal though.

**Pull request merged**